cdc/sink/manager.go

```diff
@@ -167,28 +178,43 @@ func (t *tableSink) EmitCheckpointTs(ctx context.Context, ts uint64) error {
 	return nil
 }
 
-func (t *tableSink) Close() error {
-	t.manager.destroyTableSink(t.tableID)
+// Note once the Close is called, no more events can be written to this table sink
+func (t *tableSink) Close(ctx context.Context) error {
+	return t.manager.destroyTableSink(ctx, t.tableID)
+}
+
+// Barrier is not used in table sink
+func (t *tableSink) Barrier(ctx context.Context) error {
 	return nil
 }
 
+type drawbackMsg struct {
+	tableID  model.TableID
+	callback chan struct{}
+}
+
 type bufferSink struct {
 	Sink
-	buffer chan struct {
-		rows       []*model.RowChangedEvent
-		resolvedTs model.Ts
-	}
 	checkpointTs uint64
+	buffer       map[model.TableID][]*model.RowChangedEvent
+	bufferMu     sync.Mutex
+	flushTsChan  chan uint64
+	drawbackChan chan drawbackMsg
 }
 
-func newBufferSink(ctx context.Context, backendSink Sink, errCh chan error, checkpointTs model.Ts) Sink {
+func newBufferSink(
+	ctx context.Context,
+	backendSink Sink,
+	errCh chan error,
+	checkpointTs model.Ts,
+	drawbackChan chan drawbackMsg,
+) Sink {
 	sink := &bufferSink{
-		Sink: backendSink,
-		buffer: make(chan struct {
-			rows       []*model.RowChangedEvent
-			resolvedTs model.Ts
-		}, defaultBufferChanSize),
+		Sink:         backendSink,
+		buffer:       make(map[model.TableID][]*model.RowChangedEvent),
```

Comment:
Does it have a capacity uppper limit? Will it cause OOM?

Reply:
It shares the same flow control as table sink, added a comment